### DISCUSSION TITLE
How to use a custom(?) pipeline

### DISCUSSION THREAD

**Discussion by ghost** [vladmandic/sdnext#2470] [General]
I was missing SAG in diffusers, but I found this pipeline: https://huggingface.co/docs/diffusers/api/pipelines/self_attention_guidance

Now that custom pipelines are supported, how do I actually use this?

**Comment by vladmandic** [comment]
cc @Disty0 if he can help

**Comment by Disty0** [comment]
Custom pipelines are mainly for user made pipelines like this: https://huggingface.co/Disty0/zero123plus-pipeline/tree/main

I am not sure if it will work with Diffusers native pipelines.

**Reply by ghost** [reply]
So I tried loading this pipeline
 ( https://github.com/huggingface/diffusers/blob/main/examples/community/clip_guided_stable_diffusion.py ) via the huggingface model downloader (it is one of the custom ones), but the name does not show up. I tried just searching "pipeline" and some show up but not the custom ones

What do I need to type in, and what will be the code I enter into "Custom Pipeline:" in the setting?

**Reply by Disty0** [reply]
Try typing in `clip_guided_stable_diffusion`

**Reply by ghost** [reply]
No results for this in the downloader

**Reply by Disty0** [reply]
Custom pipelines is in the Diffusers settings, not in downloader.

**Reply by ghost** [reply]
I applied this: https://s20.directupload.net/images/231110/ddbd3635.png

Then I generated some pics with SDXL base, CFG 5,7, but I can not tell if it works? It looks good but I have been exposed to some crippled generations lately so I do not know if the custom pipeline is really loaded... I downloaded nothing but I just put it in the settings like that
